mkdocs/pages/en/devbook/transactions/messages.md

```diff
@@ -0,0 +1,151 @@
+---
+title: Messages
+---
+
+# Sending Messages with Transfer Transactions
+
+<Transfer transactions:> can include an optional message field, which allows attaching up to 1,024 bytes of data to the
+transaction.
+Messages can be sent as plain text or encrypted using the recipient's public key, ensuring only the intended recipient
+can read them.
+
+This tutorial shows how to send both plain and encrypted messages, and demonstrates how to decrypt received messages.
+
+## Prerequisites
+
+If you have not done so already, start with the [Hello World](../start/hello-world.md) tutorial to make sure your
+development environment is set up correctly.
+
+Additionally, check the [Transfer transaction](transfer.md) tutorial to understand how fee
+calculation, network time, and transaction confirmation works.
+
+## Full Code
+
+{% import 'tutorial.jinja2' as tutorial with context %}
+
+{{ tutorial.code_full('devbook/transactions/messages', ['py', 'js']) }}
+
+## Code Explanation
+
+### Setting Up Accounts
+
+{{ tutorial.code_snippet(['py:14:43', 'js:10:32']) }}
+
+The tutorial uses two accounts: a sender and a recipient.
+The code reads the sender's private key from the `PRIVATE_KEY` environment variable and the recipient's public key from
+the `RECIPIENT_PUBLIC_KEY` environment variable.
+Both variables default to test keys if not set.
+
+In a real application, you typically have the recipient's address rather than their public key.
+To encrypt messages, you need to obtain their public key by querying the network using the <get:/accounts/{accountId}>
+endpoint.
+The public key is only available for accounts that have previously sent transactions.
+
+### Sending a Plain Text Message
+
+{{ tutorial.code_snippet(['py:45:108', 'js:34:91']) }}
+
+To attach a plain text message to a transfer transaction, encode the message as bytes and include it in the `message`
+field.
+
+The transaction follows the same structure as a [basic transfer transaction](transfer.md), with the addition of the
+message field:
+
+* **Message format:** Plain text messages are encoded directly as UTF-8 bytes.
+* **Maximum size:** Messages cannot exceed 1,024 bytes. The network will reject transactions with larger messages.
+
+    !!! tip "Handling larger data"
+
+        For applications requiring more than 1,024 bytes of data, consider storing the data off-chain and including a
+        hash or reference in the message field.
+        This approach maintains data integrity verification while keeping transaction sizes manageable.
+
+* **Network visibility:** Plain text messages are publicly visible on the blockchain.
+    Anyone can read them by querying the transaction.
+
+This example sends 1 XYM along with the message `"Hello, Symbol!"` to demonstrate that messages can be combined with
+mosaic transfers.
+
+!!! info "Messages without mosaics"
+
+    You can send a message without transferring any mosaics by providing an empty mosaics array.
```

Comment:
Let's reverse this. We don't send any mosaic in the code, for simplicity, and just say here that you can combine mosaic transfers and messages if you want.